File: rust/src/models/o_auth_app.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{o_auth_app_id::OAuthAppId, organization_id::OrganizationId};

/// An OAuth application registered by a platform
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct OAuthApp {
    pub client_id: String,

    pub client_secret_hint: String,

    pub created_at: String,

    pub id: OAuthAppId,

    pub is_active: bool,

    pub name: String,

    pub organization_id: OrganizationId,

    pub redirect_uris: Vec<String>,

    pub scopes: Vec<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub updated_at: Option<String>,
}

impl OAuthApp {
    pub fn new(
        client_id: String,
        client_secret_hint: String,
        created_at: String,
        id: OAuthAppId,
        is_active: bool,
        name: String,
        organization_id: OrganizationId,
        redirect_uris: Vec<String>,
        scopes: Vec<String>,
    ) -> Self {
        Self {
            client_id,
            client_secret_hint,
            created_at,
            id,
            is_active,
            name,
            organization_id,
            redirect_uris,
            scopes,
            updated_at: None,
        }
    }
}

File: typescript/src/models/oAuthApp.ts
// this file is @generated
import { parseDateTime } from "../datetime";
import { type OAuthAppId, OAuthAppIdSerializer } from "./oAuthAppId";
import { type OrganizationId, OrganizationIdSerializer } from "./organizationId";
/** An OAuth application registered by a platform */
export interface OAuthApp {
  clientId: string;

  clientSecretHint: string;

  createdAt: Date;

  id: OAuthAppId;

  isActive: boolean;

  name: string;

  organizationId: OrganizationId;

  redirectUris: string[];

  scopes: string[];

  updatedAt?: Date | null;
}

export const OAuthAppSerializer = {
  _fromJsonObject(object: any): OAuthApp {
    return {
      clientId: object["client_id"],
      clientSecretHint: object["client_secret_hint"],
      createdAt: parseDateTime(object["created_at"]),
      id: OAuthAppIdSerializer._fromJsonObject(object["id"]),
      isActive: object["is_active"],
      name: object["name"],
      organizationId: OrganizationIdSerializer._fromJsonObject(object["organization_id"]),
      redirectUris: object["redirect_uris"],
      scopes: object["scopes"],
      updatedAt:
        object["updated_at"] != null ? parseDateTime(object["updated_at"]) : undefined,
    };
  },

  _toJsonObject(self: OAuthApp): any {
    return {
      client_id: self.clientId,
      client_secret_hint: self.clientSecretHint,
      created_at: self.createdAt,
      id: OAuthAppIdSerializer._toJsonObject(self.id),
      is_active: self.isActive,
      name: self.name,
      organization_id: OrganizationIdSerializer._toJsonObject(self.organizationId),
      redirect_uris: self.redirectUris,
      scopes: self.scopes,
      updated_at: self.updatedAt,
    };
  },
};
